src/flexible_type/flexible_type_detail.hpp

```diff
@@ -535,7 +535,16 @@ struct get_int_visitor {
     return dt.posix_timestamp();
   }
   inline FLEX_ALWAYS_INLINE_FLATTEN flex_int operator()(flex_float i) const {return i; }
-  inline FLEX_ALWAYS_INLINE_FLATTEN flex_int operator()(const flex_string& t) const { return std::atoll(t.c_str()); }
+  inline FLEX_ALWAYS_INLINE_FLATTEN flex_int operator()(const flex_string& t) const { 
+    char *end;
+    long long int converted = std::strtoll(t.c_str(), &end, 10); 
+    if (*end != '\0') {
+      // was not at end of element so throw error
+      throw std::runtime_error("Invalid conversion: String contains more characters than just integer");
```

Comment:
Let's put the value into the error message, so the user has context on why it failed:
```cpp
"Invalid conversion: \"" + t + "\" cannot be interpreted as an integer."
```

Same for the float conversion below.

Reply:
That's a great idea, on it!